### PR TITLE
Crypto Analysis: Frequency Analysis 

### PR DESCRIPTION
**Pull request by HiraIjaz**

### CODE REVIEW

File: 7_Crypto Analysis.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# As output you should return a list of the normalized frequency 
+# for each of the letters a-z. 
+# The normalized frequency is simply the number of occurrences, i, 
+# divided by the total number of characters in the message, n.
+
+def freq_analysis(message):
+  def freq_analysis(message):
+  totalnum=len(message)
+  freqList=[]
+  import string
```
Comment (tasawernawaz) [inline 7_Crypto Analysis.py]: why an import statement here?

File: 7_Crypto Analysis.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#
+# The input to the function will be an encrypted body of text that only contains 
+# the lowercase letters a-z. 
+# As output you should return a list of the normalized frequency 
+# for each of the letters a-z. 
+# The normalized frequency is simply the number of occurrences, i, 
+# divided by the total number of characters in the message, n.
+
+def freq_analysis(message):
+  def freq_analysis(message):
```
Comment (tasawernawaz) [inline 7_Crypto Analysis.py]: duplicaiton?

File: 7_Crypto Analysis.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  totalnum=len(message)
```
Comment (tasawernawaz) [inline 7_Crypto Analysis.py]: `total_letters` would have been more pythonic name

**Review by tasawernawaz — CHANGES_REQUESTED** [review]

File: 7_Crypto Analysis.py (hunk 1 above)
Comment (HiraIjaz) [reply 7_Crypto Analysis.py]: i had to import string to use asciii_lowercase. but u are right i should have placed outside function body

File: 7_Crypto Analysis.py (hunk 3 above)
Comment (HiraIjaz) [reply 7_Crypto Analysis.py]: okay

File: 7_Crypto Analysis.py (hunk 2 above)
Comment (HiraIjaz) [reply 7_Crypto Analysis.py]: yes by mistake.

**Review by tasawernawaz — APPROVED** [review]

**Pull request merged**